transport/src/main/java/io/netty/channel/ChannelConfig.java

```diff
@@ -253,4 +253,12 @@
      * to detect the size of a message.
      */
     ChannelConfig setMessageSizeEstimator(MessageSizeEstimator estimator);
+
+    /**
+     * Set the {@link WriteBufferWaterMark} which is used for set the high and low
+     * water mark of the write buffer.
+     */
+    ChannelConfig setWriteBufferWaterMark(WriteBufferWaterMark writeBufferWaterMark);
```

Comment:
I had concerned about this before the pull request due to user may need to dynamic set one. but rethink again. it is reasonable to add annotation deprecated. 3ks, I will deprecated them.